src/main/java/ebay/carina/pages/desktop/ProductListingPage.java

```diff
@@ -0,0 +1,202 @@
+package ebay.carina.pages.desktop;
+
+import com.zebrunner.carina.utils.factory.DeviceType;
+import com.zebrunner.carina.webdriver.decorator.ExtendedWebElement;
+import com.zebrunner.carina.webdriver.locator.Context;
+import ebay.carina.components.common.FilterBase;
+import ebay.carina.components.desktop.FilterDesktop;
+import ebay.carina.utils.locatorenums.SortOptions;
+import ebay.carina.pages.common.LoginPageBase;
+import ebay.carina.pages.common.ProductListingPageBase;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+
+import java.math.BigDecimal;
+import java.util.LinkedList;
+import java.util.List;
+
+@DeviceType(pageType = DeviceType.Type.DESKTOP, parentClass = ProductListingPageBase.class)
+public class ProductListingPage extends ProductListingPageBase {
+    private final static Logger LOGGER = LoggerFactory.getLogger(ProductListingPage.class);
+
+    @FindBy(className = "srp-results")
+    private ExtendedWebElement divResults;
+
+    @Context(dependsOn = "divResults")
+    @FindBy(xpath = "//span[@role = 'heading']")
+    private List<ExtendedWebElement> productNameLbls;
+
+    @Context(dependsOn = "divResults")
+    @FindBy(className = "s-item__details")
+    private List<ExtendedWebElement> divProductDetails;
+    @FindBy(className = "srp-save-null-search__heading")
+    private ExtendedWebElement noProductFoundLbl;
+
+    @FindBy(xpath = "//div[@id='srp-river-results']//span[@class='s-item__price']")
+    private List<ExtendedWebElement> priceLbls;
+
+    @FindBy(id = "s0-51-16-0-1-2-6")
```

Comment:
Find another locator for such autogenerated attributes. With the next deployment of the site this locator will be changed probably